jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java

```diff
@@ -42,124 +46,109 @@ public enum ConfigWithImplementations implements Configs.Config {
-      System.setProperty("key", "systemValue");
+      System.setProperty(systemKey, systemValue);
       // TODO : Replace this when https://github.com/eclipse/jkube/issues/958 gets fixed
       final Properties fallback = new Properties();
-      fallback.put("key", "fallbackValue");
+      fallback.put(fallbackKey, fallbackValue);
       // When
       final String result = Configs.getFromSystemPropertyWithPropertiesAsFallback(fallback, "key");
       // Then
-      assertThat(result).isEqualTo("systemValue");
+      assertThat(result).isEqualTo(expected);
     } finally {
-      System.clearProperty("key");
+      System.clearProperty(systemKey);
     }
   }
 
-  @Test
-  public void getFromSystemPropertyWithPropertiesAsFallbackHasNotKeyInSystemShouldReturnSystemValue() {
-    // Given
-    try {
-      System.setProperty("not-the-key", "systemValue");
-      // TODO : Replace this when https://github.com/eclipse/jkube/issues/958 gets fixed
-      final Properties fallback = new Properties();
-      fallback.put("key", "fallbackValue");
-      // When
-      final String result = Configs.getFromSystemPropertyWithPropertiesAsFallback(fallback, "key");
-      // Then
-      assertThat(result).isEqualTo("fallbackValue");
-    } finally {
-      System.clearProperty("not-the-key");
-    }
+  @DisplayName("Config interface tests without implementation")
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("configInterfaceWithDefaultsTestData")
+  void configInterfaceWithDefaultsTest(String testDesc, String key, String expected) {
+    assertThat(key).isEqualTo(expected);
   }
 
-  @Test
-  public void getFromSystemPropertyWithPropertiesAsFallbackHasNotKeyShouldReturnNull() {
-    // Given
-    try {
-      System.setProperty("not-the-key", "systemValue");
-      // TODO : Replace this when https://github.com/eclipse/jkube/issues/958 gets fixed
-      final Properties fallback = new Properties();
-      fallback.put("not-the-key-either", "fallbackValue");
-      // When
-      final String result = Configs.getFromSystemPropertyWithPropertiesAsFallback(fallback, "key");
-      // Then
-      assertThat(result).isNull();
-    } finally {
-      System.clearProperty("not-the-key");
-    }
+  @DisplayName("Config interface tests with implementation")
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("configInterfaceWithImplementationsTestData")
+  void configInterfaceWithImplementationsTest(String testDesc, String key, String expected) {
+    assertThat(key).isEqualTo(expected);
   }
 
+  public static Stream<Arguments> propertyTestData() {
+    return Stream.of(
+            Arguments.arguments("System Property with Properties as Fallback has Key in System should return system value", "key", "systemValue", "key", "fallbackValue", "systemValue"),
+            Arguments.arguments("System Property with Properties as Fallback has not key in System should return system value", "not-the-key", "systemValue", "key", "fallbackValue", "fallbackValue"),
```

Comment:
I think there might be a typo in the description:
```suggestion
            Arguments.arguments("System Property with Properties as Fallback has not key in System should return fallback value", "not-the-key", "systemValue", "key", "fallbackValue", "fallbackValue"),
```

Reply:
Thanks for noticing this!